hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/StartupProbe.java

```diff
@@ -0,0 +1,236 @@
+                    + " retries");
+        }
+
+        // step 4: Query the mirror node gRPC API for a message on the topic created in step 2, until successful or
+        //         time runs out
+        TopicId topicId = transactionReceipt.topicId;
+        AtomicBoolean messageReceived = new AtomicBoolean();
+
+        boolean subscribedWithGrpc = false;
+	for ( retries = 1; retries <= maxRetries; retries++) {
+            try {
+                new TopicMessageQuery()
+                    .setTopicId(topicId)
+                    .subscribe(client, resp -> {
+                        messageReceived.set(true);
+                    });
+                subscribedWithGrpc = true;
+            } catch (RuntimeException e) {
+                log.warn("RuntimeException on try #{}/{} to subscribe using GRPC API call", retries, maxRetries, e);
+                // try again
+            }
+        }
+
+        if (!subscribedWithGrpc) {
+            throw new IllegalArgumentException("StartupProbe Could not successfully make GRPC API subscription call "
+                    + "after " + maxRetries + " retries");
+        }
+
+	for ( retries = 1; retries <= maxRetries; retries++) {
+            try {
+                Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+                if (messageReceived.get()) {  // we already had a previous success - no need to submit more trabsactions
+                    break;
+                }
+                new TopicMessageSubmitTransaction()
+                    .setTopicId(topicId)
+                    .setMessage("Hello, HCS! " + retries)
+                    .execute(client)
+                    .getReceipt(client);
+
+                Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+                if (Instant.now().compareTo(timeoutInstant) >= 0) {
+                    throw new TimeoutException("Ran out of time on second HAPI call - getting submit transaction "
+                            + " receipt.");
+                }
+            } catch (PrecheckStatusException pse) {
+                log.warn("PrecheckStatusException on GRPC API call to submit a transaction", pse);
+                throw new IllegalArgumentException("PrecheckStatusException on GRPC call (submitting transaction):"
+                        + pse.getMessage());
+            } catch (ReceiptStatusException rse) {
+                log.warn("ReceiptStatusException on trying to get receipt from GRPC API call", rse);
+                // try again
+            }
+        }
+
+        // make sure at least one message got through
+        if (!messageReceived.get()) {
+            throw new IllegalArgumentException("StartupProbe Could not successfully receive GRPC message after "
+                    + maxRetries + " retries");
+        }
+
+        log.info("successful for all stages - HAPI, REST API, and gRPC API."); // ideally, the only line logged.
```

Comment:
Don't need to call out all steps as that leaks implementation. Can just state:
```suggestion
        log.info("Startup probe successful");
```

Reply:
Done.